tests/Exercise/TopScorersTest.elm

```diff
@@ -0,0 +1,206 @@
+module Exercise.TopScorersTest exposing (..)
+
+import Comment exposing (Comment, CommentType(..))
+import Dict
+import Exercise.TopScorers as TopScorers
+import Review.Rule exposing (Rule)
+import Review.Test
+import Test exposing (Test, describe, test)
+import TestHelper
+
+
+rules : List Rule
+rules =
+    [ TopScorers.removeInsignificantPlayersMustUseFilter
+    , TopScorers.resetPlayerGoalCountMustUseInsert
+    , TopScorers.formatPlayersCannotUseSort
+    , TopScorers.combineGamesMustUseMerge
+    , TopScorers.aggregateScorersMustUseFoldl
+    , TopScorers.aggregateScorersMustUseUpdateGoalCountForPlayer
+    , TopScorers.formatPlayerMustUseMap
+    , TopScorers.formatPlayerMustUseWithDefault
+    ]
+
+
+failures : Test
+failures =
+    describe "Analyser"
+        [ test "should report that Dict.filter must be used" <|
+            \() ->
+                """
+module TopScorers exposing (..)
+
+removeInsignificantPlayers : Int -> Dict PlayerName Int -> Dict PlayerName Int
+removeInsignificantPlayers goalThreshold playerGoalCounts =
+    Debug.todo "implement this function"
```

Comment:
I know it's more work, but I definitely think these should be plausible solutions that students might submit instead of `Debug.todo`.
The main reason for this is a justification of why this rule exists. If we can't think of any way of writing a solution that passes the tests but break the rule, we shouldn't bother with it at all.

Reply:
Hi Jie, I do get your point here, but overall, I don't think I agree. It doesn't really matter whether or not the code solves the problem. It is the job of the exercise tests to make sure that the code solves the problem, and it is up to the analyser to check that it solves it in the right / best way.
I think the rule should fire as long as the code compiles and doesn't meet the conditions for the rule, and given that, any code that compiles is adequate.
I think the other point is that I already trust `Analyzer.functionCalls` to work, and in fact it already has plenty of checks to make sure that it does, so I don't think it make sense to retest that. I do need to test that I pass `Analyzer.functionCalls` the right arguments, but I think this is covered here.

Reply:
As an aside, I disagree that we can fully trust `Analyzer.functionCalls`, it's a complicated piece of code, and we will find bugs in it eventually. However, my point here is not that we should test it, my point is that we should test
1. The construction rule itself. For example, I just found a `Some` that should have been a `All`. This may have been found with a couple of tests.
2. Our assumption that the rule even is necessary. That can only be done by coming up with a plausible example

I'm happy to contribute to the PR and add them, if you want. As you know, I like tests :)